### PR TITLE
[extension/googleclientauth] Add new component boilerplate scaffolding

### PR DESCRIPTION
**Pull request by damemi**
**Description:**
This adds the basic component boilerplate following the recommendations for PR1 in https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/CONTRIBUTING.md#adding-new-components

**Link to tracking Issue:** https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/31412

**Testing:** Ran various `make ...` commands as listed in docs. Ignored generated files and failures from unrelated components.

**Documentation:** New readme and metadata (to be filled in, see tracking issue)

### CODE REVIEW

File: extension/googleclientauthextension/go.mod
Comment (dashpole) [inline extension/googleclientauthextension/go.mod:3]: I think this needs to be go 1.21 to match other components in the repo

File: extension/googleclientauthextension/metadata.yaml
Comment (crobert-1) [inline extension/googleclientauthextension/metadata.yaml:8]: Since this component is being added to the bug/other/feature request templates, and code owners are being added here, we probably want to add an entry to the [code owners file](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/.github/CODEOWNERS).

Comment (damemi) [reply extension/googleclientauthextension/metadata.yaml:8]: does `make update-codeowners` do that? that command fails for me btw:
```
make update-codeowners
go: cannot load module exporter/f5cloudexporter listed in go.work file: open exporter/f5cloudexporter/go.mod: no such file or directory
go: cannot load module extension/httpforwarder listed in go.work file: open extension/httpforwarder/go.mod: no such file or directory
go: cannot load module exporter/f5cloudexporter listed in go.work file: open exporter/f5cloudexporter/go.mod: no such file or directory
go: cannot load module extension/httpforwarder listed in go.work file: open extension/httpforwarder/go.mod: no such file or directory
cd cmd/githubgen && go install .
go: cannot load module ../../exporter/f5cloudexporter listed in go.work file: open ../../exporter/f5cloudexporter/go.mod: no such file or directory
go: cannot load module ../../extension/httpforwarder listed in go.work file: open ../../extension/httpforwarder/go.mod: no such file or directory
make: *** [Makefile:271: githubgen-install] Error 1
```

Comment (damemi) [reply extension/googleclientauthextension/metadata.yaml:8]: Deleted my last comment, that error was from my own go workspaces. With that updated, `make update-codeowners` still fails:

```
make update-codeowners
no Go files in /usr/local/google/home/mikedame/go/src/github.com/open-telemetry/opentelemetry-collector-contrib
cd cmd/githubgen && go install .
githubgen
2024/03/05 14:16:14 codeowners are not members: BinaryFissionGames, Frapschen, JonathanWamsley, MovieStoreGuy, Mrod1598, StefanKurek, VihasMakwana, adrielp, ag-ramachandran, akats7, alexvanboxel, antonblock, armstrmi, atingchen, atoulme, avadhut123pisal, bertysentry, braydonk, bryan-aguilar, chrroberts-pure, codeboten, cparkins, cpheps, damemi, dashpole, dehaansa, dgoscn, eedorenko, erichsueh3, evan-bradley, fatsheep9146, fchikwekwe, gjanco, greatestusername, hickeyma, hughesjj, jacobmarble, jamesmoessis, jmacd, jsuereth, kentquirk, khyatigandhi6, kkujawa-sumo, liustanley, mapno, mar4uk, marctc, martin-majlis-s1, matej-g, mcardy, moh-osman3, mrsillydog, nslaughter, pavankrish123, pcwiese, pjanotti, portertech, povilasv, psx95, pxaws, rapphil, rmfitzpatrick, rnishtala-sumo, rogercoll, samiura, schmikei, shalper2, sinkingpoint, smithclay, splunkericl, srprash, swiatekm-sumo, tamir-michaeli, tomaz-s1, varunraiko, wangzlei, yotamloe, zdaratom-s1, zpzhuSplunk
make: *** [Makefile:275: gengithub] Error 1
```

Comment (crobert-1) [reply extension/googleclientauthextension/metadata.yaml]: Thanks for pointing this out, it's fine to update code owners manually if the command fails. It looks like that failure is due to a new user being added who is working on a new component but not yet a member of the OpenTelemetry community, nothing on your end 👍

Comment (damemi) [reply extension/googleclientauthextension/metadata.yaml]: cool, thanks!

**Comment by damemi** [comment]
I tried skipping the lifecycle/shutdown tests for now in metadata.yaml and ran `mdatagen` and `make generate` but it looks like they're still running. Is there some other way I need to re-generate those tests?

**Comment by damemi** [comment]
I'm trying to run `make update-otel` to get the module on the right version  but that's just giving me a lot of this message:
```
internal/components/components.go:20:2: go.opentelemetry.io/collector/featuregate@v0.96.1-0.20240308004201-241334609fc4: invalid pseudo-version: preceding tag (v0.96.0) not found
```

**Comment by dashpole** [comment]
I would just manually update the file with the diff

File: extension/googleclientauthextension/README.md
Comment (crobert-1) [inline extension/googleclientauthextension/README.md:11]: This might be a good time to expand the README with more information. Explain the use case, define the configuration, gives examples of configurations, etc. I think this is generally part of the ["First PR" for a new component.](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/CONTRIBUTING.md#adding-new-components)

**Review by crobert-1 — COMMENTED** [review]
I'll defer to others, but it may be worthwhile to add a changelog for this. It looks like it's pretty common for other components to add a `new_component` changelog entry for even just scaffolding 👍

File: extension/googleclientauthextension/README.md
Comment (damemi) [reply extension/googleclientauthextension/README.md:11]: Sure, that doc says this further down:

>- For README.md, you can start with the following:
>```
># <Title of your component>
>
>
>```

so that's what I used, but I can add more here.

File: extension/googleclientauthextension/README.md
Comment (crobert-1) [inline extension/googleclientauthextension/README.md:43]:
```suggestion
- **project** - The Google Cloud Project telemetry is sent to if the `gcp.project.id` resource attribute is not set. If unspecified, this is determined using application default credentials.
```

File: extension/googleclientauthextension/README.md
Comment (crobert-1) [inline extension/googleclientauthextension/README.md:45]:
```suggestion
- [**quota_project**](https://cloud.google.com/apis/docs/system-parameters) - The project for quota and billing purposes. The caller must have `serviceusage.services.use` permission on the project.
```

File: extension/googleclientauthextension/README.md
Comment (crobert-1) [inline extension/googleclientauthextension/README.md:42]: Just a suggestion, it's helpful to clearly differentiate which options are required and which are default. If they're all optional (shown by the example config having no settings), maybe update the header title to make this clear.

File: extension/googleclientauthextension/README.md
Comment (crobert-1) [inline extension/googleclientauthextension/README.md:43]: Nit: For consistency, most components use `this` formatting for config option names, rather than **bold.**

**Review by crobert-1 — COMMENTED** [review]
Just some minor formatting nits. 👍  Also, I realized it might be a good time to add goleak checks to your component now, to help ensure it doesn't leak goroutines. You can simply add a `package_test.go` file to your component's directory, contents could be copied from [here.](https://github.com/open-telemetry/opentelemetry-collector/blob/main/receiver/package_test.go)

**Review by dashpole — APPROVED** [review]

File: extension/googleclientauthextension/README.md
Comment (crobert-1) [inline extension/googleclientauthextension/README.md:42]: Nit: It might make sense to have the options defined before the example. Not a big deal either way though. 👍

**Review by crobert-1 — APPROVED** [review]

**Review by andrzej-stencel — APPROVED** [review]

**Pull request merged**
